src/number_theory/cyclotomic/gal.lean

```diff
@@ -0,0 +1,183 @@
+/-
+Copyright (c) 2022 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import number_theory.cyclotomic.zeta
+import ring_theory.polynomial.cyclotomic.eval
+import field_theory.polynomial_galois_group
+/-!
+# Galois group of cyclotomic extensions
+
+In this file, we show the relationship between the Galois group of`K(ζₙ)` and `(zmod n)ˣ`;
+it is always a subgroup, and if the `n`th cyclotomic polynomial is irreducible, they are isomorphic.
+
+# Main results
+
+* `is_cyclotomic_extension.aut_to_pow_injective`: `is_primitive_root.aut_to_pow` is injective
+  in the case that it's considered over a cyclotomic field extension, where `n` does not divide
+  the characteristic of K. As a corollary, $Gal(K(ζₙ)/K)$ is abelian.
+* `is_cyclotomic_extension.aut_equiv_pow`: If, additionally, the `n`th cyclotomic polynomial is
+  irreducible in K, then `aut_to_pow` is a `mul_equiv`.
+* `gal_{X_pow/cyclotomic}_equiv_units_zmod`: Repackage `aut_to_pow` in terms of `polynomial.gal`.
+
+# References
+
+* https://kconrad.math.uconn.edu/blurbs/galoistheory/cyclotomic.pdf
+
+# TODO
+
+* `zeta_pow_power_basis` is not sufficiently general; the correct solution is some
+  `power_basis.map_conjugate`, but figuring out the exact correct assumptions + proof for this is
+  mathematically nontrivial. (Current thoughts: the ideal of polynomials with α as a root is equal
+  to the ideal of polynomials with β as a root, which, in an integrally closed domain, reduces
+  to the usual condition that they are the roots of each others' minimal polynomials.)
+
+-/
+
+local attribute [instance] pnat.fact_pos
+
+variables {L : Type*} [field L] {μ : L} (n : ℕ+) (hμ : is_primitive_root μ n)
+          (K : Type*) [field K] [algebra K L] [is_cyclotomic_extension {n} K L]
+
+local notation `ζ` := is_cyclotomic_extension.zeta n K L
+
+open polynomial ne_zero
+
+namespace is_cyclotomic_extension
+
+lemma aut_to_pow_injective [ne_zero ((n : ℕ) : K)] : function.injective $
```

Comment:
Here `L` should be explicit I think.

Reply:
I went ahead and made `L` explicit for the whole file, but I think maybe I should mess with argument order a little. Do you think `L K hµ h` (hµ a primitive root, h irreducible cyclotomic) is reasonable? Or do you think we can imply K` from irreducible cyclotomic, so skip it for those that have `h`?

Reply:
Concerning the order: if `L` is a `K`-algebra, I think `... K L` is much more natural. But `L` can be inferred from `hμ`, right? And yes, I would make `K` implicit if there is `irreducible...`

Reply:
Let's say before to be coherent with `is_cyclotomic_extension` and `zeta`.